tests/src/geometry_checker/testqgsgeometrychecks.cpp

```diff
@@ -924,7 +924,14 @@ void TestQgsGeometryChecks::testSelfContactCheck()
   QVERIFY( searchCheckErrors( checkErrors, layers["line_layer.shp"], 5, QgsPointXY( -1.2399, -1.0502 ), QgsVertexId( 0, 0, 6 ) ).size() == 1 );
   QVERIFY( searchCheckErrors( checkErrors, layers["polygon_layer.shp"], 9, QgsPointXY( -0.2080, 1.9830 ), QgsVertexId( 0, 0, 3 ) ).size() == 1 );
 
+
   cleanupTestContext( testContext );
+
+  QgsGeometryCheckContext context( 1, QgsCoordinateReferenceSystem( QStringLiteral( "EPSG:3857" ) ), QgsCoordinateTransformContext() );
```

Comment:
please add a comment about what are you testing (e.g. not empty ring), do you mind also to test if ring is empty?